nucleus/admin/launcher/src/main/java/com/sun/enterprise/admin/launcher/GFLauncher.java

```diff
@@ -660,6 +694,36 @@ private void resolveAllTokens() {
         adminFileRealmKeyFile = resolver.resolve(adminFileRealmKeyFile);
     }
 
+    private void replacePropertiesWithEnvVars(Map<String, String> properties) {
+        Pattern invalidEnvVarCharsPattern = Pattern.compile("[^_0-9a-zA-Z]");
+        for (Map.Entry<String, String> entry : properties.entrySet()) {
+            String envValue = System.getenv(entry.getKey());
+            if (envValue != null) {
+                entry.setValue(envValue);
+            } else {
+                final String sanitizedKey = invalidEnvVarCharsPattern
+                        .matcher(entry.getKey())
+                        .replaceAll("_");
+                envValue = System.getenv(sanitizedKey);
+                if (envValue != null) {
+                    entry.setValue(envValue);
+                } else {
+                    envValue = System.getenv(sanitizedKey.toUpperCase());
+                    if (envValue != null) {
+                        entry.setValue(envValue);
+                    }
+                }
+            }
+
+        }
+    }
+
+    private static Boolean isPreferEnvOverProperties(Map<String, String> properties) {
+        return Optional.ofNullable(properties.get(PREFER_ENV_VARS_OVER_PROPERTIES))
```

Comment:
Why not use `Boolean.parseBoolean()` instead?

Reply:
Of course, thanks! I forgot about this method.